custom_components/spotcast/spotcast_controller.py

```diff
@@ -117,16 +119,41 @@ def getSpotifyDeviceId(self, devices_available: dict) -> None:
 
         raise HomeAssistantError("Failed to get device id from Spotify")
 
+class TokenHelper:
+    sp_dc = None
+    sp_key = None
+    def __init__(self, sp_dc: str, sp_key: str) -> None:
+        self.sp_dc = sp_dc
+        self.sp_key = sp_key
+
+    async def start_session(self):
+        """ Starts session to get access token. """
+        cookies = { 'sp_dc': self.sp_dc, 'sp_key': self.sp_key }
+        async with aiohttp.ClientSession(cookies=cookies) as session:
+            headers = { 'user-agent': "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36" }
+            async with session.get('https://open.spotify.com/get_access_token?reason=transport&productType=web_player', allow_redirects=False, headers=headers) as response:
+                if(response.status != 200):
```

Comment:
Actually, non 200 probably is a problem (as it will probably not contain the token we need), but as with the rest of my code in this PR all the logging and safeguards are there purely to make my experimentation effective, not to work in a final solution.
Should probably do some testing for the correct content in the response (including handling empty responses).